### PR TITLE
refactor(@desktop): gif section

### LINKED ISSUE #4397: gif selector not working in base bc branch
1. enable gif in advanced settings
2. go to chat and click the gif button

result: error conatcting tenor api

### PR DESCRIPTION
**Pull request by alaibe**
Refactor gif

fixes #4397

### CODE REVIEW

**Review by jrainville — CHANGES_REQUESTED** [review]
Very nice job.

I have to ask for changes mostly because of the `GifItem` name, it's just way too confusing.

The point about adding a "real" `Item` and do the conversion in Module can be done in another PR. If you have questions about why it's better, I don't exactly remember, but it's about separation of concerns and just having an extra layer of abstraction I think. Worst case, you can ask Sale, he explains it better.

Lastly, I see that you put the Gif code in the InputArea module, which is ok and makes sense, but it adds a bit of complexity as you've seen it, since you have to pass the service to each chat section and it's a bit annoying. Anyway, all that to say that if you want, you can move the Gif module up in the architecture so that  it is not part of the chat section itself, but more an app wide module, just like I did for stickers. Unless, the gifs are gonna be different per chat, then leave it as is.

File: ui/imports/shared/status/StatusGifColumn.qml [hunk 1]
```diff
@@ -51,7 +51,7 @@ Column {
 
             StatusBaseButton {
                 id: starButton
-                property bool favorite: model.isFavorite
+                property bool favorite: RootStore.isFavorite(model.id)
```
[the remark this answers is not in the document]
Comment (alaibe) [reply ui/imports/shared/status/StatusGifColumn.qml:55]: updated

File: ui/imports/shared/status/StatusGifPopup.qml [hunk 2]
```diff
@@ -22,24 +22,22 @@ Popup {
     id: popup
     property var gifSelected: function () {}
     property var searchGif: Backpressure.debounce(searchBox, 500, function (query) {
-        // Not Refactored Yet
-//        RootStore.chatsModelInst.gif.search(query)
+        RootStore.searchGifs(query)
```
[the remark this answers is not in the document]
Comment (alaibe) [reply ui/imports/shared/status/StatusGifPopup.qml:25]: This is the shared store and this is the first component of the shared part.

File: src/app/modules/main/chat_section/module.nim [hunk 3]
```diff
@@ -39,7 +40,7 @@ proc newModule*(
     contactService: contact_service.Service,
     chatService: chat_service.Service,
     communityService: community_service.Service, 
-    messageService: message_service.Service
+    messageService: message_service.Service,
```
[the remark this answers is not in the document]
Comment (alaibe) [reply src/app/modules/main/chat_section/module.nim:43]: why not ;)

**Review by jrainville — APPROVED** [review]

File: src/app/modules/main/chat_section/chat_content/input_area/gif_column.nim [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+type
+  GifRoles {.pure.} = enum
+    Url = UserRole + 1
+    Id = UserRole + 2
+    Title = UserRole + 3
+    TinyUrl = UserRole + 4
+
+QtObject:
+  type
+    GifColumn* = ref object of QAbstractListModel
```
Comment (Khushboo-dev-cpp) [inline src/app/modules/main/chat_section/chat_content/input_area/gif_column.nim:14]: not sure but can this be renamed to something which has model in its name? easier to follow. also that how most other QAbstractListModel have been declared so far

File: ui/imports/shared/status/StatusGifPopup.qml [hunk 5]
```diff
@@ -335,36 +330,36 @@ Popup {
                 property string lastHoveredId
 
                 StatusGifColumn {
-                    // Not Refactored Yet
-//                    gifList.model: RootStore.chatsModelInst.gif.columnA
+                    gifList.model: RootStore.gifColumnA
```
Comment (Khushboo-dev-cpp) [inline ui/imports/shared/status/StatusGifPopup.qml:333]: I actually dont know what Column A B C means :(

**Review by Khushboo-dev-cpp — APPROVED** [review]

File: src/app/modules/main/chat_section/chat_content/input_area/gif_column.nim (hunk 4 above)
Comment (alaibe) [reply src/app/modules/main/chat_section/chat_content/input_area/gif_column.nim:14]: I can add model

File: ui/imports/shared/status/StatusGifPopup.qml (hunk 5 above)
Comment (alaibe) [reply ui/imports/shared/status/StatusGifPopup.qml:333]: if you open the gif popup, you will see 3 columns of gif, those are the A,B,C. The reason it is 3 models is that they can vary in height

File: src/app/modules/main/chat_section/chat_content/input_area/gif_column.nim (hunk 4 above)
Comment (Khushboo-dev-cpp) [reply src/app/modules/main/chat_section/chat_content/input_area/gif_column.nim:14]: thank you !

**Pull request merged**